src/makeInfinityAdder.js

```diff
@@ -1,10 +1,35 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let clearCall = 0;
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      clearCall++;
+
+      if (clearCall === 1) {
+        return result;
+      }
+
+      if (clearCall === 2) {
+        result = 0;
+
+        return result;
+      }
+    }
```

Comment:
The current implementation requires two calls without arguments to reset the sum which is not according to the task description. The first call without arguments should return the sum and reset it immediately after that. You need to refactor this logic to meet the requirement.